corehq/apps/reports_core/static/reports_core/js/maps.js

```diff
@@ -57,11 +57,11 @@ hqDefine('reports_core/js/maps', function () {
         if (!privates.template || !_.isEqual(config.columns, privates.columns)) {
             privates.columns = config.columns;
             var rows = _.map(privates.columns, function (col) {
-                var tr = _.template("<tr><td><%= label %></td>")(col);
-                tr += "<td><%= " + col.column_id + "%></td></tr>";
+                var tr = _.template("<tr><td><%- label %></td>")(col);
+                tr += "<td><%- " + col.column_id + "%></td></tr>";
                 return tr;
             });
-            var table = '<table class="table table-bordered"><%= rows %></table>';
+            var table = '<table class="table table-bordered"><%- rows %></table>';
```

Comment:
I think this one should not be changed. `rows` contains HTML constructed above, which includes `<tr><td>` tags that should not be escaped.

Reply:
In addition to what @millerdev said, I think the problems with this code go deeper than you gave them credit for. There are _two_ levels of injection happening here: One where they construct the _template_ itself to include arbitrary strings (e.g. `"<td><%= " + col.column_id + "%></td></tr>";`), and then another layer (the one you dealt with) where the template is rendered to insert arbitrary strings.

I took a look at the code, and I think the solution is something like this. (1) do away with string-building the template altogether and instead devise a single more complex static template that solves the problem and (2) render it with access to both privates.columns used here as well as the data row passed in in the `render` function below (not visible in this diff). Something like the following

```js
// within initPopupTemplate
...
privates.template = _.template([
    "<% _.map(columns, function (col) { %>",
    "  <tr>",
    "    <td><%- col.label %></td>",
    "    <td><%- row[col.column_id] %></td>",
    "  </tr>",
    "<% }) %>"
].join('\n'))
...
// within render
...
...(privates.template({row: row, columns: privates.columns}))...
...
```

(It's been a while. I forget if JS has better multiline string syntax than what I used there. I think `` ` ``-strings are fairly widely supported now by browsers and do allow for multi-line strings, but I forget if we use them in our codebase or not.)

Reply:
> I think this one should not be changed. rows contains HTML constructed above, which includes <tr><td> tags that should not be escaped.

@millerdev Thanks for pointing that out. I actually misunderstood the code that it is creating the table that appears below the map in the report. But it actually was for the HTML in the pop-over that comes up in the map markers (See in the image below). I will fix them and update the code.

<img width="993" alt="Screenshot 2021-04-07 at 12 05 44 PM" src="https://user-images.githubusercontent.com/7694243/113821335-975a4c80-9799-11eb-9094-5d67f954be39.png">

Reply:
Thanks @dannyroberts for the detailed explanation. I understood your point and I am aware of places where it is being used. 

> (It's been a while. I forget if JS has better multiline string syntax than what I used there. I think `-strings are fairly widely supported now by browsers and do allow for multi-line strings, but I forget if we use them in our codebase or not.)

I don't think we use them in our codebase. Template string came out in ES6. I will try to test your recommendation and will update the code accordingly.

Reply:
@dannyroberts Updated the code and tested it, seems to work fine.

<img width="400" alt="Screenshot 2021-04-07 at 12 52 01 PM" src="https://user-images.githubusercontent.com/7694243/113827107-8bbe5400-97a0-11eb-88e4-fc8ab4b0ab40.png">